### ISSUE TITLE
Hi, what is the difference on on existing module?

### ISSUE THREAD

**Issue by ekarlso** [innovationnorway/terraform-azurerm-virtual-network#1]
https://github.com/innovationnorway/terraform-azurerm-network

**Comment by joakimhellum-in** [comment]
Hi, alot of our repos are currently experimental and may be deleted any time.
Purpose of this repository is to experiment with cross-subscription hub-spoke network topology in Azure using Terraform v0.12-only features. Due to current limitation of Terraform language, this gets very complicated in pre-0.12 versions. Our Terraform configuration ends up looking like server room spaghetti and requires multiple runs in order to be able to create the networks, manage peerings and routes to central firewalls.

We would like it to look more like this potentional Terraform configuration:

```hcl
module "westeurope_network" {
  source        = "innovationnorway/network/azurerm"
  name          = "westeurope-network"
  location      = "westeurope"
  address_space = cidrsubnet(local.cidr_block, 4, 1)
  
  peerings = {
    opus = {
      remote_network_id       = module.opus_network.id
      allow_gateway_transit   = true
      allow_forwarded_traffic = true
    }
  }

  providers = {
    azurerm = "azurerm.shared"
  }
}

module "opus_network" {
  source        = "innovationnorway/network/azurerm"
  name          = "opus-network"
  location      = "westeurope"
  address_space = cidrsubnet(local.cidr_block, 4, 2)

  peerings = {
    westeurope = {
      remote_network_id       = module.opus_network.id
      use_remote_gateways     = true
      allow_forwarded_traffic = true
    }
  }

  subnets = {
    app = {
      network_number = 1
    }

    db = {
      network_number = 2
    }
  }

  routes = {
    firewall = {
      address_prefix = "0.0.0.0/0"
      next_hop_type  = "virtualappliance"
      next_hop_ip    = module.westeurope_firewall.private_ip_address
    }
  }

  providers = {
    azurerm = "azurerm.opus"
  }
}

output "opus_app_subnet_id" {
  value = module.opus.network.subnets.app.id
}
```

**Comment by ekarlso** [comment]
Yes, we have the same issue, would you like to have a chat btw ? :) Maybe we can share ideas / stuff.

**Comment by joakimhellum-in** [comment]
Yes, please :) my email is joakim.bakke.hellum@innovasjonnorge.no, also have [twitter](https://twitter.com/joakimhellum) account. We're currently working on setting up some chat rooms. I'm sure @steinpetersen-in would like to join in on as well when he has time available. We are also experimenting with cross-platform network modules for Terraform, so that we can re-use the same arguments (subnets, routes, peerings etc.) in all the major cloud providers, much like the [`terraformdns` project](https://terraformdns.github.io/about/), only for basic networking. If you are interested we woul love to discuss and share ideas about this.

**Comment by joakimhellum-in** [comment]
Hi, this repository will be deleted.
1) we will create new repos for cross-platform network modules in `terraformnetwork` organization (`terraform-azurerm-network`, `terraform-azurestack-network`, `terraform-google-network`,  `terraform-aws-network` etc.)
2) we will create new repo for Azure virtual network module that support the Azure specific features (`terraform-azurerm-virtual-network`)
